compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -2215,24 +2233,56 @@ WinogradInputTransformOp::getTiledImplementation(OpBuilder &builder,
   auto zero = builder.getIndexAttr(0);
   const int cDim = channelDim();
 
-  assert(offsets.size() == 2);
+  assert(offsets.size() == 4);
   SmallVector<OpFoldResult> inputOffsets(getInputOperandRank(), zero);
   SmallVector<OpFoldResult> outputOffsets(getOutputOperandRank(), zero);
+  const auto hDim = getImageDimensions()[0];
+  const auto wDim = getImageDimensions()[1];
   outputOffsets[2] = inputOffsets[0] = offsets[0];
-  outputOffsets[5] = inputOffsets[cDim] = offsets[1];
+  outputOffsets[3] = offsets[1];
+  outputOffsets[4] = offsets[2];
+  outputOffsets[5] = inputOffsets[cDim] = offsets[3];
 
   SmallVector<OpFoldResult> inputStrides(getInputOperandRank(), one);
   SmallVector<OpFoldResult> outputStrides(getOutputOperandRank(), one);
+  ReifiedRankedShapedTypeDims reifiedResultShapes;
+  if (failed(reifyResultShapes(builder, reifiedResultShapes))) {
+    return failure();
+  }
+  SmallVector<OpFoldResult> outputSizes = reifiedResultShapes[0];
+  SmallVector<OpFoldResult> inputSizes;
+  auto definingReifyOp =
```

Comment:
It seems like you are trying to do something here that I think can be done simpler. Some more context on this would help

Reply:
I dont think its worth being too careful here. You should be able to just query the dim of the input. Just use a method something like this
```
OpFoldResult getTensorOrMemrefDim(Value v, OpBuilder &builder, unsigned dim)
auto shapedType = v.getType().cast<ShapedType>();
auto shape = shapedType.getShape();
if (!ShapedType::isDynamic(shape[dim])) {
  return builder.getIndexAttr(shape[dim]);
}
if (isa<MemrefType>(shapedType)) {
  return builder.create<memref::DimOp>(...);
}
assert(isa<RankedTensorType>(shapedType));
return bilder.create<tensor::DimOp>(...);
```